plugins/modules/foreman_snapshot.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Manisha Singhal (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_snapshot
+short_description: Manage Foreman Snapshots
+description:
+  - "Manage Foreman Snapshots for Host Entities"
+  - "This module can create, update, revert and delete snapshots"
+  - "This module requires the foreman_snapshot_management plugin set up in the server"
+  - "See: U(https://github.com/ATIX-AG/foreman_snapshot_management)"
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+    - "apypie"
+options:
+  name:
+    description:
+      - Name of Snapshot
+    required: true
+  description:
+    description:
+      - Description of Snapshot
+    required: false
+  host:
+    description:
+      - Name of related Host
+    required: true
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: "Create a Snapshot"
+  foreman_snapshot:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_snapshot"
+    host: "host_name"
+    state: present
+
+- name: "Update a Snapshot"
+  foreman_snapshot:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_snapshot"
+    host: "host_name"
+    description: "description"
+    state: present
+
+- name: "Revert a Snapshot"
+  foreman_snapshot:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_snapshot"
+    host: "host_name"
+    state: reverted
+
+- name: "Delete a Snapshot"
+  foreman_snapshot:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_snapshot"
+    host: "host_name"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
```

Comment:
I think we can make this comment go away, especially as it's only half true now with the entity spec (it contains not only name conversions).

Reply:
Right. It is more like a relict from nailgun times, where we tried to level some inconsistencies.
Fwiw, I dropped it in the next iteration of the entity_spec pr.